src/write/unit.rs

```diff
@@ -752,6 +752,13 @@ pub enum AttributeValue {
     /// `.debug_macinfo` sections is implemented.
     DebugMacinfoRef(DebugMacinfoOffset),
 
+    /// An offset into the `.debug_macro` section.
+    ///
+    /// It is the user's responsibility to ensure the offset is valid.
```

Comment:
Maybe worth clarifying that *correctness* depends on this, but not *safety*

Reply:
Maybe something like this?

```suggestion
    /// For correctness (but not safety) it is the user's responsibility to ensure the offset is valid.
```

Reply:
I wasn't worried so much about correctness as conveying that the user was on their own here. I've updated all of these comments (this was a copy+paste from existing comments).